LanguageFeatures/Patterns/constant_A03_t02.dart

```diff
@@ -23,87 +23,80 @@
 /// literals explicitly marked const. Likewise with set and map literals versus
 /// map patterns.
 ///
-/// @description Check list, map and set literals with type argument specified
-/// in a constant patterns
+/// @description Check list, map and set literals in a constant patterns. Test
+/// if-case statement
 /// @author sgrekhov22@gmail.com
 
 // SharedOptions=--enable-experiment=patterns
 
 import "../../Utils/expect.dart";
 
 String testList(List value) {
-  switch (value) {
-    case const <int>[1, 2]:
-      return "[1, 2]";
-    case const <String>["3", "4"]:
-      return "['3', '4']";
-    case const <double>[]:
-      return "[]";
-    default:
-      return "default";
+  if (value case const [1, 2]) {
+    return "[1, 2]";
+  } else if (value case const ["3", "4"]) {
+    return "['3', '4']";
+  } else if (value case const [true, true]) {
+    return "[true, true]";
+  } else if (value case const []) {
+    return "[]";
+  } else {
+    return "default";
   }
 }
 
 String testMap(Map value) {
-  switch (value) {
-    case const <int, int>{1: 2}:
-      return "{1: 2}";
-    case const <String, num>{'answer': 42}:
-      return "{'answer': 42}";
-    case const <String, Object>{'true': true}:
-      return "{'true': true}";
-    case const <String, int>{}:
-      return "{}";
-    default:
-      return "default";
+  if (value case const {1: 2}) {
+    return "{1: 2}";
+  } else if (value case const {'answer': 42}) {
+    return "{'answer': 42}";
+  } else if (value case const {'true': true}) {
+    return "{'true': true}";
+  } else if (value case const {}) {
+    return "{}";
+  } else {
+    return "default";
   }
 }
 
 String testSet(Set value) {
-  switch (value) {
-    case const <num>{1, 2, 3}:
-      return "{1, 2, 3}";
-    case const <String>{'1', '2', '3'}:
-      return "{'1', '2', '3'}";
-    case const <double>{}:
-      return "{}";
-    default:
-      return "default";
+  if (value case const {1, 2, 3}) {
+    return "{1, 2, 3}";
+  } else if (value case const {'1', '2', '3'}) {
+    return "{'1', '2', '3'}";
+  } else if (value case const {}) {
+    return "{}";
+  } else {
+    return "default";
   }
 }
 
 main() {
   Expect.equals("[1, 2]", testList(const [1, 2]));
```

Comment:
In this case the type argument for the constant list is definitely `dynamic`.

Reply:
@eernstg  please clarify how exactly this test should be rewritten? I'm confused because now (`Dart SDK version: 2.19.0-374.0.dev (dev) (Fri Nov 4 12:40:18 2022 -0700) on "windows_x64"`) we have the following behavior.
```dart
String test(List value) {
  switch (value) {
    case const [1, 2]:
      return "[1, 2]";
    default:
      return "default";
  }
}

String testDynamic(List value) {
  switch (value) {
    case const <dynamic>[1, 2]:
      return "<dynamic>[1, 2]";
    default:
      return "default";
  }
}

String testInt(List value) {
  switch (value) {
    case const <int>[1, 2]:
      return "<int>[1, 2]";
    default:
      return "default";
  }
}

main() {
  Expect.equals("[1, 2]", test(const [1, 2])); // Ok
  Expect.equals("<dynamic>[1, 2]", testDynamic(const [1, 2])); // Ok
  Expect.equals("<int>[1, 2]", testInt(const [1, 2])); // Fails
}
```
But in case of `if-case` statement we have
```dart
String test(List value) {
  if (value case const [1, 2]) {
    return "[1, 2]";
  } else {
    return "default";
  }
}

String testDynamic(List value) {
  if (value case const <dynamic>[1, 2]) {
    return "<dynamic>[1, 2]";
  } else {
    return "default";
  }
}

String testInt(List value) {
  if (value case const <int>[1, 2]) {
    return "<int>[1, 2]";
  } else {
    return "default";
  }
}

main() {
  Expect.equals("[1, 2]", test(const [1, 2])); // fail
  Expect.equals("<dynamic>[1, 2]", testDynamic(const [1, 2])); // fail
  Expect.equals("<int>[1, 2]", testInt(const [1, 2])); // fail
}
```
What is wrong and what is correct here?